tests/integration/targets/mso_remote_location/tasks/main.yml

```diff
@@ -0,0 +1,267 @@
+
+- name: Test that we have an ACI MultiSite host, username and password
+  fail:
+    msg: 'Please define the following variables: mso_hostname, mso_username and mso_password.'
+  when: mso_hostname is not defined or mso_username is not defined or mso_password is not defined
+
+# CLEAN ENVIRONMENT
+
+- name: Set vars
+  set_fact: 
+    mso_info: &mso_info
+      host: '{{ mso_hostname }}'
+      username: '{{ mso_username }}'
+      password: '{{ mso_password }}'
+      validate_certs: '{{ mso_validate_certs | default(false) }}'
+      use_ssl: '{{ mso_use_ssl | default(true) }}'
+      use_proxy: '{{ mso_use_proxy | default(true) }}'
+      output_level: '{{ mso_output_level | default("info") }}'
+
+- name: Query all backups
+  cisco.mso.mso_backup:
+    <<: *mso_info
+    state: query
+  delegate_to: localhost
+  register: backups
+
+- name: Ensure all backups with link to remote location ansible_test are removed
+  cisco.mso.mso_backup:
+    <<: *mso_info
+    backup: "{{ item.name }}"
+    state: absent
+  when:
+    - item.remoteLocationName is defined
+    - item.remoteLocationName == "ansible_test"
+  loop: "{{ backups.current }}"
+  delegate_to: localhost
+
+- name: Ensure remote location ansible_test is removed
+  cisco.mso.mso_remote_location:
+    <<: *mso_info
+    remote_location: ansible_test
+    state: absent
+  delegate_to: localhost
+
+- name: Configure remote location scp (check mode)
+  cisco.mso.mso_remote_location: &remote_location
+    <<: *mso_info
+    remote_location: ansible_test
+    remote_protocol: scp
+    remote_host: 173.36.219.25
+    remote_path: '/tmp'
+    remote_authentication_type: password
+    remote_username: '{{ mso_username }}'
+    remote_password: '{{ mso_password }}'
+    state: present
+  check_mode: yes
+  delegate_to: localhost
+  register: cm_config_remote
+
+- name: Configure remote location scp
+  cisco.mso.mso_remote_location:
+    <<: *remote_location
+  delegate_to: localhost
+  register: nm_config_remote
+
+- name: Verify configuration
+  assert:
+    that:
+      - cm_config_remote is changed
+      - cm_config_remote.current.name == "ansible_test"
+      - cm_config_remote.current.credential.authType == "password"
+      - cm_config_remote.current.credential.port == 22
+      - cm_config_remote.current.credential.protocolType == "scp"
+      - cm_config_remote.current.credential.hostname == "173.36.219.25"
+      - cm_config_remote.current.credential.remotePath == "/tmp"
+      - cm_config_remote.current.credential.username == '{{ mso_username }}'
+      - nm_config_remote is changed
+      - nm_config_remote.current.name == "ansible_test"
+      - nm_config_remote.current.credential.authType == "password"
+      - nm_config_remote.current.credential.port == 22
+      - nm_config_remote.current.credential.protocolType == "scp"
+      - nm_config_remote.current.credential.hostname == "173.36.219.25"
```

Comment:
check against the variable not the actual IP.

Reply:
ok